pkg/templates/notificationtemplates/notification_generator.go

```diff
@@ -1 +1,84 @@
 package notificationtemplates
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("notification-templates")
+var templates map[string]NotificationTemplate
+
+// NotificationTemplate contains the template subject and content
+type NotificationTemplate struct {
+	Subject string
+	Content string
+}
+
+type Option func(asset *Assets)
+
+func WithAssets(a Assets) Option {
+	return func(assets *Assets) {
+		templates = nil
+		assets.names = a.names
+		assets.asset = a.asset
+	}
+}
+
+func WithEmptyCache() Option {
+	return func(_ *Assets) {
+		templates = nil
+	}
+}
+
+// GetNotificationTemplates returns a notification subject and body or an error
```

Comment:
could you also mention the meaning of the returned `bool` value in the description?

Reply:
done here: 41d79047577d9c630ddcfa119727316fa607bf84